test/Mock/Datasets/ShipmentResponses.php

```diff
@@ -0,0 +1,531 @@
+                'number' => $testData['number'] ?? '31',
+                'person' => $testData['person'] ?? 'Test Person',
+                'company' => $testData['company'] ?? 'MyParcel',
+                'email' => $testData['email'] ?? 'test@myparcel.nl',
+                'phone' => $testData['phone'] ?? '0612345678',
+            ],
+            'options' => [
+                'signature' => $testData['signature'] ?? false,
+                'insurance' => [
+                    'amount' => $testData['insurance'] ?? 0,
+                    'currency' => 'EUR'
+                ],
+                'return' => $testData['return'] ?? false,
+                'package_type' => $testData['package_type'] ?? 1,
+            ],
+        ]);
+    }
+    
+    /**
+     * bpost specific response set
+     */
+    public static function getBpostFlow(array $testData = []): array
+    {
+        // Default to BE address, but use NL address if country is NL
+        $isNL = ($testData['country'] ?? 'BE') === 'NL';
+        $defaultStreet = $isNL ? 'Antareslaan' : 'Adriaan Brouwerstraat';
+        $defaultNumber = $isNL ? '31' : '16';
+        $defaultPostalCode = $isNL ? '2132JE' : '2000';
+        $defaultCity = $isNL ? 'Hoofddorp' : 'Antwerpen';
+        
+        return self::getStandardShipmentFlow([
+            'reference_identifier' => $testData['reference_identifier'] ?? null,
+            'carrier_id' => 2, // bpost correct ID
+            'recipient' => [
+                'cc' => $testData['country'] ?? 'BE',
+                'postal_code' => $testData['postal_code'] ?? $defaultPostalCode,
+                'city' => $testData['city'] ?? $defaultCity,
+                'street' => $testData['street'] ?? $defaultStreet,
+                'number' => $testData['number'] ?? $defaultNumber,
+                'person' => $testData['person'] ?? 'Test Person',
+                'company' => $testData['company'] ?? 'MyParcel',
+                'email' => $testData['email'] ?? 'test@myparcel.nl',
+                'phone' => $testData['phone'] ?? '0612345678',
+            ],
+            'options' => [
+                'signature' => $testData['signature'] ?? false,
+                'insurance' => [
+                    'amount' => $testData['insurance'] ?? 0,
+                    'currency' => 'EUR'
+                ],
+                'return' => $testData['return'] ?? false,
+                'package_type' => $testData['package_type'] ?? 1,
+                'delivery_type' => $testData['delivery_type'] ?? 2,
+                'only_recipient' => $testData['only_recipient'] ?? false,
+            ],
+        ]);
+    }
+    
+    /**
+     * UPS specific response set
+     */
+    public static function getUPSFlow(array $testData = [], int $carrierId = 14): array
+    {
+        $country = $testData['country'] ?? 'DE';
+        $defaultStreet = $country === 'DE' ? 'Feldstrasse' : ($country === 'NL' ? 'Hoofdstraat' : 'Antareslaan');
```

Comment:
This is not going to scale well (what if we add a testcase for FR?) and may be fixed by adding a logic functionality / class to handle the $testData arrays as mentioned earlier.